crates/cubecl-runtime/src/memory_management/mod.rs

```diff
@@ -79,3 +79,12 @@ pub struct MemoryDeviceProperties {
     /// The required memory offset alignment in bytes.
     pub alignment: u64,
 }
+
+/// Properties of the device related to topology.
+#[derive(Debug, Clone)]
+pub struct TopologyProperties {
+    /// The minimum size of a subcube on this device
+    pub subcube_size_min: u32,
+    /// The maximum size of a subcube on this device
+    pub subcube_size_max: u32,
+}
```

Comment:
Is the first dimension in the Cube responsible for the subcube size? For example, if I have an AMD GPU with a max of 64 and a min of 32, and I launched a kernel with (32, 8, 1), would I have 8 subcubes of 32, or 4 of 64? It would be great to add the answer here as a comment.

Reply:
I don't believe there's an actual answer that's documented anywhere. In HIP it's a compiler flag that determines it, wgpu is still very in flux and not very well defined.  For Intel, it depends on the registers. Really, it's quite unreliable and the only definitive number is the one fetched at runtime. But this at least provides a rough estimate for determining things like cube size, with the exact details left to each kernel.